e2e/playwright/test-utils.ts

```diff
@@ -0,0 +1,114 @@
+import { expect, Page } from '@playwright/test'
+import { EngineCommand } from '../../src/lang/std/engineConnection'
+
+async function waitForPageLoad(page: Page) {
+  // wait for 'Loading KittyCAD Modeling App...' spinner
+  await page.getByTestId('initial-load').waitFor()
+  // wait for 'Loading stream...' spinner
+  await page.getByTestId('loading-stream').waitFor()
+  // wait for all spinners to be gone
+  await page.getByTestId('loading').waitFor({ state: 'detached' })
+
+  await page.getByTestId('start-sketch').waitFor()
+}
+
+async function removeCurrentCode(page: Page) {
+  const hotkey = process.platform === 'darwin' ? 'Meta' : 'Control'
+  await page.click('.cm-content')
+  await page.keyboard.down(hotkey)
+  await page.keyboard.press('a')
+  await page.keyboard.up(hotkey)
+  await page.keyboard.press('Backspace')
+  await expect(page.locator('.cm-content')).toHaveText('')
+}
+
+async function sendCustomCmd(page: Page, cmd: EngineCommand) {
+  await page.fill('[data-testid="custom-cmd-input"]', JSON.stringify(cmd))
+  await page.click('[data-testid="custom-cmd-send-button"]')
+}
+
+async function clearCommandLogs(page: Page) {
+  await page.click('[data-testid="clear-commands"]')
+}
+
+async function expectCmdLog(page: Page, locatorStr: string) {
+  await expect(page.locator(locatorStr)).toBeVisible()
+}
+
+async function waitForDefaultPlanesToBeVisible(page: Page) {
+  await page.waitForFunction(
+    () =>
+      document.querySelectorAll('[data-receive-command-type="object_visible"]')
+        .length >= 3
+  )
```

Comment:
I couldn't figure out how to make a locator that waited for three of the same selector to appear, so kept this for the time being.